### PR TITLE
Add support for post-install/remove scripts and custom deb packages

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This provides features necessary for dotnet/sdk Linux packages.

### To double check:

* [ ] The right tests are in and the right validation has happened.  Guidance: https://github.com/dotnet/arcade/blob/main/Documentation/Validation.md

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets [hunk 1]
```diff
@@ -247,6 +247,26 @@
...
+          SkipUnchangedFiles="True"
+          UseHardlinksIfPossible="True" />
+
+    <Copy SourceFiles="@(LinuxPostRemoveScript)"
+          DestinationFiles="@(_ControlFile->'$(_LayoutControlRoot)postrm')"
+          OverwriteReadOnlyFiles="True"
+          SkipUnchangedFiles="True"
+          UseHardlinksIfPossible="True" />
+
+    <Exec Command="chmod 0755 '$(_LayoutControlRoot)postinst' '$(_LayoutControlRoot)postrm'" />
```
Comment (NikolaMilosavljevic) [inline src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets:268]: Is this equivalent to SDK's `ugo+x` - https://github.com/dotnet/sdk/blob/04f5de5910f0aed55161b40fbbbe4dbfdbbd2a2c/src/Installer/redist-installer/targets/GenerateDebs.targets#L284-L286

Comment (jkoritzinsky) [reply src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets:268]: This is equivalent to `rwx-rw-rw`, so not quite the same as that case. I'll change it to be consistent.

File: src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets [hunk 2]
```diff
@@ -247,6 +247,26 @@
       <FileWrites Include="@(_ControlFile)" />
     </ItemGroup>
 
+    <Copy SourceFiles="@(DebControlFile)"
+          DestinationFiles="@(_ControlFile->'$(_LayoutControlRoot)%(Filename)%(Extension)')"
```
Comment (NikolaMilosavljevic) [inline src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets:251]: Should this be? `DestinationFiles="@(DebControlFile->'$(_LayoutControlRoot)%(Filename)%(Extension)')"`

Comment (jkoritzinsky) [reply src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets:251]:
```suggestion
    <Copy SourceFiles="@(DebControlFile)"
          DestinationFiles="@(DebControlFile->'$(_LayoutControlRoot)%(Filename)%(Extension)')"
```

File: src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets [hunk 3]
```diff
@@ -247,6 +247,26 @@
...
     </ItemGroup>
 
+    <Copy SourceFiles="@(DebControlFile)"
+          DestinationFiles="@(DebControlFile->'$(_LayoutControlRoot)%(Filename)%(Extension)')"
+          OverwriteReadOnlyFiles="True"
+          SkipUnchangedFiles="True"
+          UseHardlinksIfPossible="True" />
+
+    <Copy SourceFiles="@(LinuxPostInstallScript)"
+          DestinationFiles="@(_ControlFile->'$(_LayoutControlRoot)postinst')"
```
Comment (NikolaMilosavljevic) [inline src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.build.targets:257]: Same question to the one I had for `Copy` task right above. Is `DestinationFiles` value correct? Same question for the `Copy` task right below.

**Review by NikolaMilosavljevic — APPROVED** [review]

**Pull request merged**